File: libs/monobitmap-manager/src/test/kotlin/mono/bitmap/manager/factory/TextBitmapFactoryTest.kt
/*
 * Copyright (c) 2023, tuanchauict
 */

package mono.bitmap.manager.factory

import mono.graphics.geo.Rect
import mono.shape.shape.Text
import kotlin.test.Test
import kotlin.test.assertEquals

/**
 * A test for [TextBitmapFactory]
 */
class TextBitmapFactoryTest {
    @Test
    fun testToBitmap() {
        val text = Text(Rect.byLTWH(0, 0, 7, 5))
        text.setText("012345678\nabc")
        val bitmap = TextBitmapFactory.toBitmap(
            text.bound.size,
            text.renderableText.getRenderableText(),
            text.extra,
            isTextEditingMode = false
        )
        assertEquals(
            """
                |┌─────┐
                |│01234│
                |│5678 │
                |│ abc │
                |└─────┘
            """.trimMargin(),
            bitmap.toString()
        )
    }
}

File: monosketch-svelte/src/lib/mono/monobitmap/manager/text-bitmap-factory.test.ts
/*
 * Copyright (c) 2024, tuanchauict
 */

import { Rect } from "$libs/graphics-geo/rect";
import { StringExt } from "$libs/string";
import { describe, expect, test } from "vitest";
import { TextBitmapFactory } from './text-bitmap-factory';
import { Text } from '$mono/shape/shape/text';
import trimIndent = StringExt.trimIndent;

describe('TextBitmapFactory', () => {
    test('toBitmap', () => {
        const text = Text.fromRect({ rect: Rect.byLTWH(0, 0, 7, 5) });
        text.setText('012345678\nabc');
        const bitmap = TextBitmapFactory.toBitmap(
            text.bound.size,
            text.renderableText.getRenderableText(),
            text.extra,
            false,
        );
        expect(bitmap.toString().trim()).toBe(
            trimIndent(`
            ┌─────┐
            │01234│
            │5678 │
            │ abc │
            └─────┘
            `),
        );
    });
});
